pallets/parachain-staking/src/lib.rs

```diff
@@ -137,53 +136,128 @@ pub mod pallet {
 	/// Snapshot of collator state at the start of the round for which they are selected
 	pub struct CollatorSnapshot<AccountId, Balance> {
 		pub bond: Balance,
-		pub nominators: Vec<Bond<AccountId, Balance>>,
+		pub delegations: Vec<Bond<AccountId, Balance>>,
 		pub total: Balance,
 	}
 
 	#[derive(Encode, Decode, RuntimeDebug, TypeInfo)]
-	/// Collator state with commission fee, bonded stake, and nominations
+	/// DEPRECATED
+	/// Collator state with commission fee, bonded stake, and delegations
 	pub struct Collator2<AccountId, Balance> {
 		/// The account of this collator
 		pub id: AccountId,
 		/// This collator's self stake.
 		pub bond: Balance,
 		/// Set of all nominator AccountIds (to prevent >1 nomination per AccountId)
 		pub nominators: OrderedSet<AccountId>,
-		/// Top T::MaxNominatorsPerCollator::get() nominators, ordered greatest to least
+		/// Top T::MaxDelegatorsPerCollator::get() nominators, ordered greatest to least
 		pub top_nominators: Vec<Bond<AccountId, Balance>>,
 		/// Bottom nominators (unbounded), ordered least to greatest
 		pub bottom_nominators: Vec<Bond<AccountId, Balance>>,
-		/// Sum of top nominations + self.bond
+		/// Sum of top delegations + self.bond
 		pub total_counted: Balance,
-		/// Sum of all nominations + self.bond = (total_counted + uncounted)
+		/// Sum of all delegations + self.bond = (total_counted + uncounted)
 		pub total_backing: Balance,
 		/// Current status of the collator
 		pub state: CollatorStatus,
 	}
 
-	/// Convey relevant information describing if a nominator was added to the top or bottom
-	/// Nominations added to the top yield a new total
+	impl<A, B> From<Collator2<A, B>> for CollatorCandidate<A, B> {
+		fn from(other: Collator2<A, B>) -> CollatorCandidate<A, B> {
+			CollatorCandidate {
+				id: other.id,
+				bond: other.bond,
+				delegators: other.nominators,
+				top_delegations: other.top_nominators,
+				bottom_delegations: other.bottom_nominators,
+				total_counted: other.total_counted,
+				total_backing: other.total_backing,
+				request: None,
+				state: other.state,
+			}
+		}
+	}
+
+	#[derive(PartialEq, Clone, Copy, Encode, Decode, RuntimeDebug, TypeInfo)]
+	/// Changes allowed by an active collator candidate to their self bond
+	pub enum CandidateBondChange {
+		Increase,
+		Decrease,
+	}
+
+	#[derive(PartialEq, Clone, Copy, Encode, Decode, RuntimeDebug, TypeInfo)]
+	/// Request scheduled to change the collator candidate self-bond
+	pub struct CandidateBondRequest<Balance> {
+		pub amount: Balance,
+		pub change: CandidateBondChange,
+		pub when: RoundIndex,
```

Comment:
`when` is a bit ambiguous (especially as I read more of the code), maybe a name that makes it clear whether it means "when scheduled" or "when eligible" would help

Reply:
TODO: change to `when_executable`

I feel like `when_scheduled` implies automatic execution, but it is not scheduled in that way.

Reply:
@4meta5 reminder there is a todo

Reply:
done